### ISSUE TITLE
Build errors after upgrading to Next 13.4.5

### ISSUE THREAD

**Issue by mortennajbjergjp** [thebuilder/react-intersection-observer#628]
I get a build error in my Next/React project after upgrading NextJS to 13.4.5.

```
Failed to compile.

./node_modules/.pnpm/react-intersection-observer@9.4.4_react@18.2.0/node_modules/react-intersection-observer/react-intersection-observer.modern.mjs
Attempted import error: 'Component' is not exported from 'react' (imported as 'React').
```

I can solve the issue by changing this:

`import * as React from 'react';`

into:

`import React from 'react';`

In the `react-intersection-observer.modern.mjs` file.

Also (though not directly related to this issue) adding a "use client" directive seems to be necessary for me.

**Comment by thebuilder** [comment]
Any idea what can cause the `import` issue?
`import * as React from 'react';` should be the correct way of importing React, with the other being old/deprecated.

**Comment by mortennajbjergjp** [comment]
> Can you try out the beta release? It should support the "use client"
> 
> ```shell
> npm install react-intersection-observer@beta
> ```

Great Daniel. Thank you for your quick response.
I tried the beta release and can confirm that this release solves both of my issues.

I have no clue as to why the import suddenly throws errors in that version of NextJS. I peeked in the release logs but there's a lot going on between the releases so I couldn't track it down.

Thanks!

**Comment by thebuilder** [comment]
It's a bit scary switching out the bundling logic. Never know if it breaks something in some weird setups out there. Seems to be working fine so far.

**Comment by thebuilder** [comment]
Fixed by #629
